libkbfs/folder_branch_ops.go

```diff
@@ -548,9 +548,21 @@ func (fbo *folderBranchOps) setHeadLocked(
 		// revision is enough to trigger conflict resolution.
 		fbo.cr.Resolve(md.Revision(), MetadataRevisionUninitialized)
 	} else if md.MergedStatus() == Merged {
-		// If we are already merged through this write, the revision would be the
-		// latestMergedRevision on server.
-		fbo.setLatestMergedRevisionLocked(ctx, lState, md.Revision(), false)
+		journalEnabled := TLFJournalEnabled(fbo.config, fbo.id())
+		var key kbfscrypto.VerifyingKey
+		if journalEnabled {
+			key, err = fbo.config.KBPKI().GetCurrentVerifyingKey(ctx)
+			if err != nil {
+				return err
+			}
+		}
+		// If this is a merged revision, it should be the latest one
+		// we know about.  However, if journaling is enabled, and this
```

Comment:
This function (`setHeadLocked`) doesn't only get called when processing journal entries, it gets called either 1) when the local device does a write, or 2) when we fetch the next update from the server.  We still need to set the latest merged revision in the latter case.